retrieval-api/src/save-provider-retrieval/index.ts

```diff
@@ -0,0 +1,60 @@
+import { PostgresProviderRetrievalRepo } from "../infra/repos/postgres-provider-retrieval-repo";
+import { ProviderRetrieval } from "../models/provider-retrieval";
+import { ValidationError } from "../models/validation-error";
+import { ProviderRetrievalRepo } from "../repos/provider-retrieval-repo";
+
+class SaveProviderRetrieval {
+  private providerRetrievalRepo: ProviderRetrievalRepo;
+
+  constructor(props: SaveProviderRetrievalProps) {
+    this.providerRetrievalRepo = props.providerRetrievalRepo;
+  }
+
+  async execute(request: SaveProviderRetrievalDto): Promise<void> {
+    try {
+      await this.providerRetrievalRepo.save(request.providerRetrieval)
+    } catch(err) {
+      console.error("Could not save provider retrieval.");
+      throw err;
+    }
+  }
+}
+
+class SaveProviderRetrievalController {
```

Comment:
this appears to be a duplicate of the code in `retrieval-api/src/infra/controllers/save-provider-retrieval-controller.ts`. Am I missing something?